tools/src/main/java/org/nmdp/ngs/tools/ValidateInterpretation.java

```diff
@@ -47,105 +49,118 @@
-            Map<String, SubjectTyping> observed = readObserved(observedFile,lociList);	
-   
-            for (String sample : expected.keySet()) {
-            	
-            	if(observed.get(sample) != null){        		
-            		
-            		for(String loc : lociList){
-
-		                List<String> alleles = expected.get(sample).getTyping(loc);
-		                List<String> interpretations = observed.get(sample).getTyping(loc);
-	
-		                if(interpretations.size() != 0){
-			                for (String expectedAllele : alleles) {
-			                    boolean result = false;
-			                    
-			                    for (String interpretation : interpretations) {
-		                  	
-			                        List<String> interpretedAlleles = Splitter.onPattern("[/|]+").splitToList(interpretation);
-			                        List<String> found = new ArrayList<String>();
-			                                     
-			                        for (String interpretedAllele : interpretedAlleles) {
-			                            if (matchByField(expectedAllele, interpretedAllele) >= resolution) {
-			                                found.add(interpretedAllele);
-			                            }                            
-			                        }
-			
-			                        if (!found.isEmpty()) {
-			                            result = true;
-			                        }
-			                    }
-			                    
-	
-			                    if (result) {
-			                        passes++;
-			                    }
-			                    else {
-			                        failures++;
-			                    }
-			
-			                    if (!printSummary) {
-			                        writer.println((result ? "PASS" : "FAIL") + "\t" + sample + "\t" + expectedAllele + "\t" + expected.get(sample).getExperiment());
-			                    }
-			                }
-	            		}
-            		}
-            	}
-            }
 
+            // interpretations keyed by sample
+            ListMultimap<String, Interpretation> expected = read(expectedFile);
+            ListMultimap<String, Interpretation> observed = read(observedFile);
+
+            // for each sample in observed
+            for (String sample : observed.keySet()) {
+                for (Interpretation o : observed.get(sample)) {
+                    AlleleList observedAlleleList = asAlleleList(o);
+                    if (shouldValidate(o, observedAlleleList)) {
+
+                        // for each matching sample in expected
+                        for (Interpretation e : expected.get(sample)) {
+                            Genotype expectedGenotype = asGenotype(e);
+                            for (Haplotype expectedHaplotype : expectedGenotype.getHaplotypes()) {
+                                for (AlleleList expectedAlleleList : expectedHaplotype.getAlleleLists()) {
```

Comment:
Yes.  There is a lot of complexity (e.g. haplotypes) that is glossed over currently.